proposals/eligibility_validator.go

```diff
@@ -140,7 +142,8 @@ func (v *Validator) CheckEligibility(ctx context.Context, ballot *types.Ballot)
 		beacon,
 	)
 
-	v.beacons.ReportBeaconFromBallot(epoch, ballot.ID(), beacon, weight)
+	weightPer := fixed.DivUint64(atxWeight, uint64(numEligibleSlots))
+	v.beacons.ReportBeaconFromBallot(epoch, ballot, beacon, weightPer)
```

Comment:
what if ballots is with several eligibilities?

Reply:
this number is multiplied by number of eligibilities here https://github.com/spacemeshos/go-spacemesh/blob/e9d1a564bda11964b9a49f3d820757cd7989a140/beacon/beacon.go#L260